common/types/mapper/thrift/shared.go

```diff
@@ -1044,7 +1044,7 @@ func FromDecisionTaskCompletedEventAttributes(t *types.DecisionTaskCompletedEven
 		ScheduledEventId: t.ScheduledEventID,
 		StartedEventId:   t.StartedEventID,
 		Identity:         &t.Identity,
-		BinaryChecksum:   t.BinaryChecksum,
+		BinaryChecksum:   &t.BinaryChecksum,
```

Comment:
Bad binaries are [checked](https://github.com/uber/cadence/blob/master/service/frontend/workflowHandler.go#L574) on `PollForDecisionTask` request. If client don't send this field (`nil`) we treat it as empty string and pass it to the check function. It then checks against domain config for bad binaries.

So, client that do not set this field will be allowed. Unless `""` binary is explicitly configured as bad binary, then it would be denied. It seems, that empty binary is valid, I see no validations checks for that.